python-package/lightgbm/sklearn.py

```diff
@@ -460,24 +467,12 @@ def fit(self, X, y,
             feval = _eval_function_wrapper(eval_metric)
         else:
             feval = None
-            # register default metric for consistency with callable eval_metric case
```

Comment:
> I think this change has backward compatibility because default value for metric is set in c++ as the same way as the current code.

Yeah, but I'm talking about the Python side (we had many issues about inconsistent behavior at Python side and this code solved them). So, this code should be leaved as is to prevent the future changes in current behavior. Please see this our old conversation (setting default metric only at cpp side **is not enough**): https://github.com/Microsoft/LightGBM/pull/1589#discussion_r211144307

Reply:
@henry0312 Please fix this (leave old code) to get this PR merged.
I think that in case of we don't introduce new arguments (please see https://github.com/Microsoft/LightGBM/pull/1912#pullrequestreview-187054280), it's enought just to add something like this after the old code:
```
if params['train_metric'] or params['valid_metric']:
    params['metric'] = None
```

Reply:
@henry0312 @guolinke I've decided to dive into the "metrics hell" again and started to write a big test which reflects the current behavior of metrics handling: https://github.com/Microsoft/LightGBM/commit/826546941dcaf9f700ff471fde0567d41347e55a ([`metrics_test` branch](https://github.com/Microsoft/LightGBM/compare/metrics_test)). The next step is sklearn - the hardest part. After this, I'll create a PR.

Reply:
PR has been created: #1954.

@henry0312  I hope it will help you abandon the idea to create more arguments for metrics 😄 
I like the idea of separate metrics a lot, but seriously, corresponding keys in `params` is quite enough.